### PR TITLE
Viet/redeemer exunits and attach

### PR DESCRIPTION
**Pull request by vvtran**
https://github.com/Plutonomicon/cardano-browser-tx/issues/167

### CODE REVIEW

File: src/Types/ScriptLookups.purs [hunk 1]
```diff
@@ -301,29 +289,23 @@ instance Semigroup ValueSpentBalances where
...
   , valueSpentBalancesOutputs :: ValueSpentBalances
   -- Balance of the values produced and required for the transaction's outputs
   , datums :: Array Datum
-  -- *Set* of accumulating datums so we can use `setScriptDataHash`
-  , redeemers :: Map RedeemerTag (Array Redeemer)
-  -- Map of accumulating redeemers separated by `RedeemerTag`. Note that adding
-  -- to this discards the ordering of redeemers from constraints (should that
-  -- matter).
+  -- Ordered accumulation of datums so we can use to `setScriptDataHash`
+  , redeemers :: Array (RedeemerTag /\ T.Redeemer)
```
Comment (ngua) [inline src/Types/ScriptLookups.purs:298]: I can't think of why we might need the `RedeemerTag`. Is there a specific future scenario you're thinking of? These are only being stored for `setScriptDataHash`, right? We always need to include all redeemers when calling that (AFAIK), so I don't think we'll need to distinguish between different redeemer types

File: src/PlutusData.purs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module PlutusData
```
Comment (ngua) [inline src/PlutusData.purs:1]: Could this go into `Helpers`? If so I personally think that would be better than a whole new module for this (unless there's some circular dependency issue with that too)

**Review by ngua — COMMENTED** [review]
Looks good, a few comments below

File: src/Types/ScriptLookups.purs (hunk 1 above)
Comment (vvtran) [reply src/Types/ScriptLookups.purs:298]: Sure, I can remove.

File: src/PlutusData.purs (hunk 2 above)
Comment (vvtran) [reply src/PlutusData.purs:1]: Will try add it in.

Comment (vvtran) [reply src/PlutusData.purs:1]: Hmm, that caused some horrible cyclic deps.

Comment (vvtran) [reply src/PlutusData.purs:1]: Going to leave it for now.

Comment (ngua) [reply src/PlutusData.purs:1]: OK, no problem. There's no way around the extra module then

**Review by ngua — APPROVED** [review]
I _think_ this is the correct way to deal with the redeemer pointers. We will find out definitively when we start submitting smart contract transactions

**Pull request merged**